lib/mongoid/clients/storage_options.rb

```diff
@@ -7,9 +7,10 @@ module StorageOptions
 
       included do
 
-        cattr_accessor :storage_options, instance_writer: false do
-          storage_options_defaults
-        end
+        # NOTE: use class_attribute here so that when the child modifies its
+        # storage options, the parent is not also modified. Using cattr_accessor
+        # causes both to be modified
+        class_attribute :storage_options, instance_writer: false, default: storage_options_defaults
```

Comment:
use class_attribute here so that when the child modifies its storage options, the parent is not also modified. Using cattr_accessor causes both to be modified.